<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>SpatialExperiment_Bioc2021 â€¢ SpatialExperimentBioc2021</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="SpatialExperiment_Bioc2021">
<meta property="og:description" content="Cells spatial organization within tissues is important for the 
    understanding of cell communications and interactions in developmental, 
    physiological, and pathological states.
    Lately several technologies are emerging for the joint extraction of
    transcripts quantification and their spatial organization at single cell level.
    Among others, seqFISH and 10x Visium Spatial Gene Expression (VSGE) seem to be 
    the most widely used platforms because of their ability to extract hundreds to 
    thousands of transcripts at the same time in the first case and for the standard 
    protocol implemented in the latter.
    Briefly, seqFISH combines temporal barcodes in multiple hybridization rounds 
    with microscopy fluorescent imaging to detect the spatial coordinates of
    transcripts.
    On the other hand, VSGE allows the joint analysis of microscopy images of a
    chip-placed frozen piece of tissue and its transcriptome at spot-barcode level.
    Given the lack of dedicated general tools for spatial transcriptomics, 
    we developed the SpatialExperiment Bioconductor package, which presents a
    set of two dedicated classes (SpatialExperiment and VisiumExperiment) both 
    extending the SingleCellExperiment Bioconductor class with slots and methods 
    for spatial transcriptomics data handling.
    This work could become the basis for facilitating further development of
    methods aimed to analyze this type of data, e.g., for spatial clustering 
    and the identification of spatially variable genes.
    Here, we present a short workshop showing how to analyze spatial transcriptomics
    with our SpatialExperiment package and how to use it in combination with other 
    analysis software.
    We will start by showing how to combine a SpatialExperiment with a 
    SingleCellExperiment to store multimodal datasets such as seqFISH 
    and scRNAseq and how to manage them to graphically visualize the data.
    Additionally, we will show how to create a VisiumExperiment object 
    starting from 10x Genomics public datasets and how to manage it for 
    clustering annotation and plotting. ">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">SpatialExperimentBioc2021</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/Introduction.html">1. Introduction to Spatial Transcriptomics</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/drighelli/SpatialExperiment_Bioc2021">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">

<div id="bioc2021-spatialexperiment-workshop" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#bioc2021-spatialexperiment-workshop" class="anchor"></a>Bioc2021 SpatialExperiment Workshop</h1></div>
<p>This three-parts workshop provides basics backgroud and the instruments to start to work with spatial transcriptomics with R/Bioconductor.</p>
<p>The first part is an overview of the most widely used technologies for spatially resolved transcriptomics, such as seqFISH and 10x Genomics Visium, and the main differences between them.</p>
<p>The second part is mainly focused on the <code>SpatialExperiment</code> class and how to handle its methods for storing and retrieving spatial coordinates, images, and how to manage them.</p>
<p>The third part will show how to store/retrieve already stored spatial datasets from the <code>STexampleData</code> and <code>10xVisiumData</code> packages and how to generate plots with <code>ggspavis</code>.</p>
<p>The workshop will end with a mini-challenge for the attendees to handle with provided tools during the sessions.</p>
</div>
<div id="authors" class="section level1">
<h1 class="hasAnchor">
<a href="#authors" class="anchor"></a>Authors:</h1>
<ul>
<li>
<a href="github.com/drighelli">Dario Righelli</a> (<a href="mailto:dario.righelli@gmail.com" class="email">dario.righelli@gmail.com</a>)</li>
<li>
<a href="github.com/HelenaLC">Helena L. Crowell</a> (<a href="mailto:helena.crowell@uzh.ch" class="email">helena.crowell@uzh.ch</a>)</li>
<li>
<a href="https://lmweber.org/">Lukas M. Weber</a> (<a href="mailto:lukas.weber@jhu.edu" class="email">lukas.weber@jhu.edu</a>)</li>
</ul>
</div>
<div id="pre-requisites" class="section level1">
<h1 class="hasAnchor">
<a href="#pre-requisites" class="anchor"></a>Pre-requisites</h1>
<ul>
<li>Basic R syntax knowledge and R data structures</li>
<li>Familiarity with <a href="https://bioconductor.org/packages/SingleCellExperiment/">SingleCellExperiment</a> and/or <a href="https://bioconductor.org/packages/SummarizedExperiment/">SummarizedExperiment</a> classes</li>
</ul>
</div>
<div id="available-resources" class="section level1">
<h1 class="hasAnchor">
<a href="#available-resources" class="anchor"></a>Available Resources</h1>
<ul>
<li>Link to the <code>pkgdown</code> <a href="https://drighelli.github.io/SpatialExperiment_Bioc2021/">website</a>.</li>
<li>Link to <a href="https://hub.docker.com/r/drighelli/spatialexperiment_bioc2021">Docker image</a>.</li>
<li>The <code>SpatialExperiment</code> <a href="https://bioconductor.org/packages/SpatialExperiment/">link</a>.</li>
<li>The <code>STexampleData</code> package <a href="https://bioconductor.org/packages/STexampleData">link</a>.</li>
<li>The <code>TENxVisiumData</code> package <a href="https://bioconductor.org/packages/TENxVisiumData">link</a>.</li>
<li>The <code>ggspavis</code> package <a href="https://github.com/lmweber/ggspavis">link</a>.</li>
</ul>
<div id="to-use-the-docker-image" class="section level2">
<h2 class="hasAnchor">
<a href="#to-use-the-docker-image" class="anchor"></a>To use the Docker image:</h2>
<p>We highly suggest to use the docker image within all the needed material.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode sh"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="ex">docker</span> run <span class="at">-e</span> PASSWORD=<span class="op">&lt;</span>SET_A_PASSWORD<span class="op">&gt;</span> -p 8787:8787 drighelli/spatialexperiment_bioc2021</span></code></pre></div>
<p>Once running, navigate to <a href="https://localhost:8787/" class="uri">https://localhost:8787/</a> and then login with <code>rstudio</code>:<code>&lt;YourChosenPassword&gt;</code>.</p>
</div>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Browse source code at <br><a href="https://github.com/drighelli/SpatialExperiment_Bioc2021/">https://â€‹github.com/â€‹drighelli/â€‹SpatialExperiment_Bioc2021/â€‹</a>
</li>
<li>Report a bug at <br><a href="https://github.com/drighelli/SpatialExperiment_Bioc2021/issues/new/">https://â€‹github.com/â€‹drighelli/â€‹SpatialExperiment_Bioc2021/â€‹issues/â€‹new/â€‹</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Dario Righelli <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0003-1504-3583" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Helena L Crowell <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-4801-1767" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Lukas M Weber <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-3282-1730" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p>Developed by Dario Righelli, Helena L Crowell, Lukas M Weber.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
