---
title: "3. The STexampleData package"
author:
  - name: "Lukas M. Weber"
    affiliation: "Johns Hopkins Bloomberg School of Public Health, Baltimore, USA"
    email: "lukas.weber@jhu.edu"
output: rmarkdown::html_vignette
vignette: >
    %\VignetteIndexEntry{3. The STexampleData package}
    %\VignetteEncoding{UTF-8}
    %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  chunk_output_type: console
---

# STexampleData package

The [STexampleData](http://bioconductor.org/packages/STexampleData) package provides access to several spatially resolved transcriptomics datasets, which have been formatted into the [SpatialExperiment](https://bioconductor.org/packages/SpatialExperiment) Bioconductor class.

These datasets have been collected from various publicly available sources, and cover several technological platforms.

We provide these datasets as `SpatialExperiment` objects to make them easier to access, so that we and others can use them for examples, demonstrations, tutorials, and other purposes.

Additional datasets from the 10x Genomics Visium platform are also available in the [TENxVisiumData](https://bioconductor.org/packages/TENxVisiumData) package (developed by Helena Crowell).


# Datasets

The package currently contains the following datasets:

- `Visium_humanDLPFC` (10x Genomics Visium): A single sample (sample 151673) of human brain dorsolateral prefrontal cortex (DLPFC) in the human brain, measured using the 10x Genomics Visium platform. This is a subset of the full dataset containing 12 samples from 3 neurotypical donors, published by [Maynard and Collado-Torres et al. (2021)](https://www.nature.com/articles/s41593-020-00787-0). The full dataset is available from the [spatialLIBD](http://bioconductor.org/packages/spatialLIBD) Bioconductor package.

- `Visium_mouseCoronal` (10x Genomics Visium): A single coronal section from the mouse brain, spanning one hemisphere. This dataset was previously released by 10x Genomics on their [website](https://support.10xgenomics.com/spatial-gene-expression/datasets/1.1.0/V1_Adult_Mouse_Brain).

- `seqFISH_mouseEmbryo` (seqFISH): A subset of cells (embryo 1, z-slice 2) from a previously published dataset investigating mouse embryogenesis by [Lohoff and Ghazanfar et al. (2020)](https://www.biorxiv.org/content/10.1101/2020.11.20.391896v1), generated using the seqFISH platform. The full dataset is [available online](https://marionilab.cruk.cam.ac.uk/SpatialMouseAtlas/).


# Install package

The package can be installed from Bioconductor as follows.

```{r, message=FALSE}
# BiocManager::install("STexampleData", version = "devel")
```

Alternatively, the latest version can also be installed from GitHub. For this workshop, we use the latest development version from GitHub.

```{r, message=FALSE}
# remotes::install_github("lmweber/STexampleData")
```


# Load dataset

First, we check the available datasets from the `STexampleData` package stored in Bioconductor's ExperimentHub resource.

```{r, message=FALSE}
# load packages
library(ExperimentHub)
```

```{r}
# query ExperimentHub resource
eh <- ExperimentHub()
query(eh, "STexampleData")
```


Next, we run a short example showing how to load one of the datasets from the `STexampleData` package.

Alternative ways of loading the datasets from ExperimentHub are shown in the package vignette.

```{r, message=FALSE}
# load packages
library(SpatialExperiment)
library(STexampleData)
```

```{r}
# load dataset from latest development version
spe <- load_data("Visium_humanDLPFC")

# alternatively: load dataset from ExperimentHub
# spe <- Visium_humanDLPFC()
```


Then, we have a look at the structure of the `SpatialExperiment` object.

```{r}
spe
dim(spe)
assayNames(spe)
rowData(spe)
colData(spe)
spatialData(spe)
head(spatialCoords(spe))
imgData(spe)
```


# Plot dataset

Next, we generate some simple plots of the dataset. We can use plotting functions available in the [ggspavis](https://github.com/lmweber/ggspavis) package, which is currently available from GitHub.

```{r, message=FALSE}
# install ggspavis package
# remotes::install_github("lmweber/ggspavis")
```

```{r, message=FALSE}
# load packages
library(ggspavis)
```


Create some plots of the dataset in x-y coordinates. This dataset contains a set of manually annotated ground truth labels for cortical layers, which we can use to display annotation on the plot.

```{r, out.width="65%"}
# plot spots only
plotSpots(spe)
```

```{r, out.width="65%"}
# plot spots with annotation
plotSpots(spe, annotate = "ground_truth")
```


We can also use a different plotting function to display the spots and annotation along with the histology image as background, which is also stored in the `SpatialExperiment` object. We also highlight spots located over tissue, which are identified with the "in_tissue" column in the `spatialData`.

```{r, out.width="65%"}
# plots with histology background
plotVisium(spe)
plotVisium(spe, fill = "ground_truth", highlight = "in_tissue")
```


# Exercise

For some additional practice with these objects, attempt the following:

- Generate a plot in x-y coordinates with annotation showing total unique molecular identifier (UMI) counts
- Generate a plot in x-y coordinates with annotation showing expression of a specific gene of interest, e.g. SNAP25, MOBP, or PCP4

You can refer to the help files for the plotting functions with `?plotSpots` and `?plotVisium`.

For a further exercise, try adjusting the color palettes and other plotting options, as described in the help files.

<!-- Possible solution 1 -->
<!-- colData(spe)$sum <- colSums(counts(spe)) -->
<!-- plotVisium(spe, fill = "sum") -->

<!-- Possible solution 2 -->
<!-- id <- rowData(spe)$gene_id[which(rowData(spe)$gene_name == "PCP4")] -->
<!-- plotVisium(spe, fill = id, assay = "counts") -->

