---
title: "2. The SpatialExperiment class"
author:
    - name: "Helena L. Crowell"
      affiliation: "Department of Molecular Life Sciences, University of Zurich, Zurich, Switzerland"
      email: "helena.crowell@uzh.ch"
output: rmarkdown::html_vignette
vignette: >
    %\VignetteIndexEntry{2. The SpatialExperiment class}
    %\VignetteEngine{knitr::rmarkdown}
    %\VignetteEncoding{UTF-8}
---

<style type="text/css"> .smaller { font-size: 10px } </style>

# Reading 10X Visium data

The 10X Genomics' *CellRanger* pipeline will process data using standard output file formats that are saved, for each sample, in a single directory `/<sample>/outs/` of the following structure:

```{bash eval = FALSE}
sample
|—outs 
··|—raw/filtered_feature_bc_matrix.h5
··|—raw/filtered_feature_bc_matrix
····|—barcodes.tsv
····|—features.tsv
····|—matrix.mtx
··|—spatial
····|—tissue_hires_image.png
····|—tissue_lowres_image.png
····|—detected_tissue_image.jpg
····|—aligned_fiducials.jpg
····|—scalefactors_json.json
····|—tissue_positions_list.csv
```

The `SpatialExperiment` package provides an exemplary 10X Visium spatial gene expression data of two serial mouse brain sections (Sagittal-Posterior) available from the [10X Genomics website](https://support.10xgenomics.com/spatial-gene-expression/datasets). These are located in the `extdata/10xVisium` directory:

```{r message = FALSE, warning = FALSE}
library(SpatialExperiment)

dir <- system.file(
  file.path("extdata", "10xVisium"),
  package = "SpatialExperiment")

sample_ids <- c("section1", "section2")
samples <- file.path(dir, sample_ids)
```

We can load these data into a `SpatialExperiment` using the `read10xVisium()` function, which will read in all relevant information, including the count data, spatial metadata and coordinates, scale factors, and images:

```{r}
list.files(samples[1])
list.files(file.path(samples[1], "spatial"))

(spe <- read10xVisium(samples, sample_ids,
  type = "sparse",   # use sparse (not HDF5) format
  data = "raw",      # read all (not filtered) data
  images = "lowres", # specify which image(s) to include
  load = TRUE))      # specify whether or not to load image(s)
```

# The `SpatialExperiment` class

## `spatialData` & `-Coords`

Spatial metadata are stored inside the `int_colData`'s `spatialData` field as a `DataFrame` and include:

* `in_tissue` indicating whether an observation was mapped to tissue
* `array_row/col` giving the spots' row/column coordinate in the array[^1]

[^1]: `array_row`s range from 0-77 (78 rows); `array_col`s are even in 0-126 for even rows, and odd in 1-127 for odd rows (64 columns), giving in $78 \times 64 = 4,992$ spots per sample.

```{r}
head(spatialData(spe))
# tabulate number of spots mapped to tissue
table(
    sample_id = spe$sample_id,
    in_tissue = spatialData(spe)$in_tissue)
```

A numeric matrix of spatial coordinates (here, x and y) is stored inside the `int_colData`'s `spatialCoords` field, and can be accessed using the `spatialCoords()` accessor:

```{r}
head(spatialCoords(spe))
```

Note that the `colData`, `spatialData`, and `spatialCoords` slots follow a hierarchical structure where `colData` > `spatialData` > `spatialCoords`. Here, each accessor function allows joint accession of the target slot, and (optionally) any slot(s) that precedes it.

Specifically, the following commands are supported may be used to access specific subsets of (spatial) metadata associated with each column (observation, e.g. spots or cells) in a SPE:

```{r results = "hide"}
spatialCoords(spe)

spatialData(spe)
spatialData(spe, spatialCoords = TRUE)

colData(spe, spatialData = TRUE)
colData(spe, spatialCoords = TRUE)
colData(spe, spatialData = TRUE, spatialCoords = TRUE)
```

## `imgData`

Image-related data are stored in the `int_metadata`'s `imgData` field as a `DataFrame` with the following columns:

* `sample_id` and `image_id` specifying the image's sample and image identifier
* `data` is a list of `SpatialImage`s containing the image's `grob`, path and/or URL
* `scaleFactor`s are used to rescale spatial coordinates according to the image's resolution

We can retrieve these data using the `imgData()` accessor:

```{r}
imgData(spe)
```

# The `SpatialImage` class

Images inside a `SpatialExperiment`'s `imgData` are stored as objects of class `SpatialImage`. The `SpatialImage` class hierarchy provides representations of images from a variety of sources, including:

* `LoadedSpatialImage`: an image that is fully realized into memory, where the raster representation is stored inside the output object.
* `RemoteSpatialImage`: an image that is remotely hosted and retrieved only on request.
* `StoredSpatialImage`: an image that is stored in a local file and is loaded into memory only on request.

A list of `SpatialImage`s can be retrieved from the `imgData`'s `data` field using the `$` accessor:

```{r}
imgData(spe)$data
```

For `StoredSpatialImage` and `RemoteSpatialImage` objects, loading the image with `imgRaster()` (see next section) will automatically store the loaded `raster` object in an in-memory cache. Any subsequent `imgRaster()` call will retrieve the `raster` from the cache, avoiding costly retrieval from the file system.

This provides the option to store an image's source at minimal storage cost, which is desirable when multiple images are to be stored (say, for many samples and of different resolutions), or when a `SpatialExperiment` is to be exported.

## Methods for image handling

The `SpatialExperiment` package provides various functions to handle which and how image data is stored in the object. These include:

* `get/add/rmvImg` to retrieve/add/remove (an) image(s) from a `SpatialExperiment`
* `imgSource/Raster` to retrieve the path/URL or `raster` object associated with (an) image(s)

`get/rmvImg()` and `imgSource/Raster()` are flexible in the specification of the `sample/image_id` arguments. Specifically,

- `TRUE` is equivalent to *all*, e.g. `sample_id = "<sample>", image_id = TRUE` will drop all images for a given sample.
- `NULL` defaults to the first entry available, e.g., `sample_id = "<sample>", image_id = NULL` will drop the first image for a given sample.

For example, `sample_id,image_id = TRUE,TRUE` will specify all images; `NULL,NULL` corresponds to the first image entry in the `imgData`; `TRUE,NULL` equals the first image for all samples; and `NULL,TRUE` matches all images for the first sample.

### `raster` & source accession

Data available in an object of class `SpatialImage` may be accessed via the `imgRaster()` and `imgSource()` accessors. Note that, since our images are `LoadedSpatialImage`s, there source is `NA`:

```{r}
si <- imgData(spe)$data[[1]]
head(imgRaster(si)[, 1:4])
imgSource(si)
```

Using `as.raster()` or, equivalently, the `imgRaster()` accessor, a `SpatialImage` may be converted to a `raster` object, which can be used directly for plotting:

```{r fig.small = TRUE}
img1 <- as.raster(si)
img2 <- imgRaster(si)
identical(img1, img2)
par(mar = c(0,0,0,0))
plot(img1)
```

In the example below, we access *all* images as `raster` objects (via setting `sample_id = image_id = TRUE`):

```{r}
imgs <- imgRaster(spe, sample_id = TRUE, image_id = TRUE)
sapply(imgs, \(.) c(dim = dim(.), class = class(.)))
```

### Adding & removing images

Besides a path or URL to source the image from and a numeric scale factor, `addImg()` requires specification of the `sample_id` the new image belongs to, and an `image_id` that is not yet in use for that sample:

```{r fig.small=TRUE}
url <- "https://i.redd.it/3pw5uah7xo041.jpg"
spe <- addImg(spe,
    sample_id = "section1", 
    image_id = "pomeranian",
    imageSource = url, 
    scaleFactor = NA_real_, 
    load = TRUE)
```

The above code chunk has added an new image entry in the input `SpatialExperiment`'s `imgData` field:

```{r fig.small = TRUE}
imgData(spe)
img <- imgRaster(spe,
    sample_id = "section1",
    image_id = "pomeranian")
plot(img)
```

We can remove specific images with `rmvImg()`:

```{r}
spe <- rmvImg(spe,
    sample_id = "section1",
    image_id = "pomeranian")
imgData(spe)
```

# `sample_id` replacement

While storing of `sample_id`s inside the `SpatialExperiment`'s `colData` enables direct accessibility via the `colData` and `$` accessors, this field is protected against arbitrary modification. This affects operations to the following effects:

Empty replacement will retain sample identifiers:

```{r}
tmp <- spe$sample_id
spe$sample_id <- NULL
identical(tmp, spe$sample_id)
```

Replacement of `sample_id`s is permitted provided that

i. the number of unique sample identifiers is retained
ii. newly provided sample identifiers are a one-to-one mapping

```{r error = TRUE}
spe$sample_id <- sample(c("a", "b", "c"), ncol(spe), TRUE)
spe$sample_id <- sample(c("a", "b"), ncol(spe), TRUE)
```

Valid replacement will be propagated to the `imgData`:

```{r}
tmp <- spe
i <- as.numeric(factor(spe$sample_id))
tmp$sample_id <- c("sampleA", "sampleB")[i]
imgData(tmp)
```

# Session info {.smaller}

```{r session-info}
sessionInfo()
```
