<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>1. Introduction to Spatial Transcriptomics • SpatialExperimentBioc2021</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="1. Introduction to Spatial Transcriptomics">
<meta property="og:description" content="SpatialExperimentBioc2021">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SpatialExperimentBioc2021</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Introduction.html">1. Introduction to Spatial Transcriptomics</a>
    </li>
    <li>
      <a href="../articles/STexampleData.html">3. The STexampleData package</a>
    </li>
    <li>
      <a href="../articles/SpatialExperiment.html">2. The SpatialExperiment class</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/drighelli/SpatialExperiment_Bioc2021">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="Introduction_files/header-attrs-2.9/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>1. Introduction to Spatial Transcriptomics</h1>
                        <h4 class="author">Dario Righelli</h4>
            <address class="author_afil">
      Department of Statistical Sciences, University of Padova, Padua, Italy<br><a class="author_email" href="mailto:#"></a><a href="mailto:dario.righelli@unipd.it" class="email">dario.righelli@unipd.it</a>
      </address>
                  
      
      <small class="dont-index">Source: <a href="https://github.com/drighelli/SpatialExperiment_Bioc2021/blob/master/vignettes/Introduction.Rmd"><code>vignettes/Introduction.Rmd</code></a></small>
      <div class="hidden name"><code>Introduction.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>Spatial organization of cells in tissues are well known to be particularly relevant to understand their functions, which in turn are influenced by transcripts expression and epigenomics factor. To unveil this relation, in the last years, multiple technologies are emerging to allow to extract this information. We refer to these technologies with <strong>Spatially Resolved Technologies</strong>, but during this workshop we’ll <strong>focus</strong> on Spatial <strong>Transcriptomics</strong> Technologies.</p>
</div>
<div id="spatial-transcriptomics-technologies" class="section level1">
<h1 class="hasAnchor">
<a href="#spatial-transcriptomics-technologies" class="anchor"></a>Spatial Transcriptomics Technologies</h1>
<p>With Spatial Transcriptomics (ST) we refer to all those technologies able to <strong>joint</strong> detect <strong>transcripts quantification</strong> with their <strong>spatial coordinates</strong> into the understudy tissue.</p>
<p>At the moment, we distinguish two main families of ST, <strong>molecular-based</strong> and <strong>spot-based</strong> technologies.</p>
<div id="molecular-based-family" class="section level2">
<h2 class="hasAnchor">
<a href="#molecular-based-family" class="anchor"></a>Molecular-Based family</h2>
<p>The main idea of these technologies is to detect the transcripts locations in a tissue in <strong>multiple temporal snapshots</strong> at <strong>cellular</strong> and <strong>sub-cellular</strong> level.</p>
<p>Examples of this family are: <em>seqFISH(+)</em>, <em>MERFISH</em>, others.</p>
<p>Main <strong>differences</strong> within this family are due to the way of <em>transcripts coordinates</em> are detected and the <em>number of detected transcripts</em>.</p>
<div id="the-seqfish" class="section level3">
<h3 class="hasAnchor">
<a href="#the-seqfish" class="anchor"></a>The seqFISH(+)</h3>
<p>seqFISH technology (Lubeck et al. 2014) uses <strong>temporal barcodes</strong> of transcripts <strong>multiple hybridization rounds</strong> in combination with <strong>image microscopy</strong> and <strong>FISH</strong> approach.</p>
<p><br></p>
<p>This allows detecting the localization of the transcripts at each barcoded time.</p>
<p><br></p>
<p><img src="https://images.squarespace-cdn.com/content/v1/5a4ac56051a584ab2ccb89b0/1516120774943-7WPM1PIY31CZOCV7SGD6/ke17ZwdGBToddI8pDm48kBAGMl9lAFNwmUiPbCJ2eMkUqsxRUqqbr1mOJYKfIPR7LoDQ9mXPOjoJoqy81S2I8N_N4V1vUb5AoIIIbLZhVYxCRW4BPu10St3TBAUQYVKcn-QjMN7HnL4Bymikkk2WTfFgLouyrutj2pCFluUg_HsXmCHOHkitiJoryoUeM1GK/seqFISH-Barcoding.PNG" alt="Hybridization seqFISH" width="500"></p>
<p><br></p>
<p>For each Barcode we have multiple hybridization rounds to co-localize transcripts.</p>
<p>N.B. <strong>No cDNA or PCR amplification</strong> is needed with this technology.</p>
<p><br></p>
<p><img src="https://images.squarespace-cdn.com/content/v1/5a4ac56051a584ab2ccb89b0/1516134310807-RMZ7EHRQP9Z13QPL21S4/ke17ZwdGBToddI8pDm48kNacjlcq_7nfuzlN6oepL9IUqsxRUqqbr1mOJYKfIPR7LoDQ9mXPOjoJoqy81S2I8N_N4V1vUb5AoIIIbLZhVYxCRW4BPu10St3TBAUQYVKc1sOXCvrBwV_7t32kh__lTviCsFUeUHYeUGwMyWpfSTHO-EhU1b7j_5Efn46yFIvZ/seqFISH-Single_Molecule_Sensitivity.PNG" alt="tissue snapshots" width="500"></p>
<p><br></p>
<p>This process can be directly applied on a tissue slice.</p>
<p><br></p>
<p><img src="https://images.squarespace-cdn.com/content/v1/5a4ac56051a584ab2ccb89b0/1516127183330-JXQ47UO5EAS4D694WLDZ/ke17ZwdGBToddI8pDm48kPDOvGbVLEL9nOU6oZxVbzwUqsxRUqqbr1mOJYKfIPR7LoDQ9mXPOjoJoqy81S2I8N_N4V1vUb5AoIIIbLZhVYy7Mythp_T-mtop-vrsUOmeInPi9iDjx9w8K4ZfjXt2dhxdRMEtDg5mXKZgN2Fd53Y5VUjyZdXk48oecIvq7S57CjLISwBs8eEdxAxTptZAUg/seqFISH_preserves_spatial_context_in_tissues.png" alt="tissue snapshots" width="500"></p>
<p><br></p>
<p>The first version (<em>seqFISH</em>) of this technology allowed to detect lower number of transcripts, while the second version (<em>seqFISH+</em>) (Eng et al. 2019) allows detecting thousands of transcripts at each round which allows this technology to be comparable to a whole transcriptome one.</p>
</div>
</div>
<div id="spot-based-technologies" class="section level2">
<h2 class="hasAnchor">
<a href="#spot-based-technologies" class="anchor"></a>Spot-Based technologies</h2>
<p>The main idea of this technology is to provide uniquely identified (i.e. barcoded) <strong>reference spots</strong> with associated <strong>coordinates</strong> and <strong>transcripts abundance</strong>. The number of cells in the spot depends on the spot size.</p>
<p>Technology examples of this family are: <em>10x Visium Spatial Gene Expression</em>, <em>Slide-seq(V2)</em>.</p>
<p>Main <strong>differences</strong> within this family are due to spots size and the spots reference protocol.</p>
<div id="the-10x-genomics-visium-spatial-gene-expression" class="section level3">
<h3 class="hasAnchor">
<a href="#the-10x-genomics-visium-spatial-gene-expression" class="anchor"></a>The 10x Genomics Visium Spatial Gene Expression</h3>
<p>This technology uses ad-hoc designed <strong>barcode-spots based chips</strong> where <strong>tissue slices</strong> can be placed on.</p>
<p>The number of spots is pre-defined, so we already know how many spots are on the rows and columns of the chip.</p>
<p>Here is an example of a <strong>Mouse Brain tissue slice</strong> placed on a 10x Visium chip.</p>
<!-- <br> -->
<!-- <img src="https://github.com/drighelli/EuroBioc2020_SpatialWorkshop/raw/master/inst/imgs/visium_tissue_lowres_image.png" alt="10x visium mouse brain" width="500"/> -->
<!-- <br> -->
<p><br></p>
<p><img src="https://github.com/drighelli/SpatialExperiment_Bioc2021/raw/master/inst/imgs/visium_detected_tissue_image.jpg" alt="10x visium mouse brain" width="500"></p>
<p><br></p>
<p>While we take a snapshot in full resolution of the on-chip placed tissue slice, the unique barcoded-spots <strong>do cDNA amplification</strong> of the transcripts.</p>
<p>In such a way, at the end, we have an <strong>image reference</strong> of the experiment and a <strong>whole transcriptome</strong> amplification per each spot.</p>
<p>(Details about the experiment output will be given during the second part of this workshop by <em>Helena</em> and an example dataset will be provided during the third part by <em>Lukas</em>.)</p>
</div>
</div>
</div>
<div id="the-spatialexperiment-class" class="section level1">
<h1 class="hasAnchor">
<a href="#the-spatialexperiment-class" class="anchor"></a>The SpatialExperiment class</h1>
<p>To facilitate the implementation of methodologies and spatial data handling in R, we designed a new class named <a href="https://bioconductor.org/packages/SpatialExperiment/"><code>SpatialExperiment</code></a>. This class inherits from the popular Bioconductor class <a href="https://bioconductor.org/packages/SingleCellExperiment"><code>SingleCellExperiment</code></a> and extends it with specific slots and get/set methods for spatial data.</p>
<p>Here is a schematics of the organization of the actual (devel version) <code>SpatialExperiment</code> class.</p>
<p><br><img src="https://github.com/drighelli/SpatialExperiment_Bioc2021/raw/master/inst/imgs/spe.png" alt="The SpatialExperiment class" width="500"></p>
<p><br></p>
<p>The class is organized around the <strong>assays</strong> data structure (more details in part 2):</p>
<p>Attributes:</p>
<ul>
<li>
<em>assays</em>: contains the primary and/or transformed data (such as counts matrix)</li>
<li>
<em>rowData</em>: information about the assays rows*</li>
<li>
<em>colData</em>: information about the assays columns* and <em>spatial coordinates</em>
</li>
<li>
<em>reducedDims</em>: dimention reductions representations (such as PCA/tSNE/UMAP)</li>
<li>
<em>imgData</em>: samples related images.</li>
</ul>
<p>SpatialExperiment proper Methods:</p>
<ul>
<li>
<em>spatialCoords</em>: returns/sets the spatial coordinates in a matrix form</li>
<li>
<em>spatialData</em>: returns the spatial metadata</li>
<li>
<em>imgData</em>: gets/sets the imgData</li>
<li>
<em>loadImg, unloadImg, addImg, removeImg</em>: imgData handling methods</li>
<li>
<em>spatialCoordsNames</em>: returns the name of stored spatial coordinates</li>
<li>
<em>spatialDataNames</em>: returns the spatialData colnames</li>
<li>
<em>molecules</em>: dedicated assay for molecule-based data, based on BumpyMatrix class</li>
</ul>
<p>The <em>SpatialExperiment</em> class for this workshop is the official Bioconductor <em>devel</em> version of it.</p>
<p>To simplify your analysis today, you can use the <strong>Docker image</strong> simply by running:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="ex">docker</span> run <span class="at">-e</span> PASSWORD=<span class="op">&lt;</span>SET_A_PASSWORD<span class="op">&gt;</span> -p 8787:8787 drighelli/spatialexperiment_bioc2021</span></code></pre></div>
<p>Once docker command finished download and installation, navigate to <a href="https://localhost:8787/">https://localhost:8787/</a> and then login with rstudio:<yourchosenpassword>.</yourchosenpassword></p>
</div>
<div id="in-short" class="section level1">
<h1 class="hasAnchor">
<a href="#in-short" class="anchor"></a>In short</h1>
<p>We provided the main overview of Spatial Transcriptomics technologies, focusing on seqFISH for the molecule-based family and on 10x Genomics Visium Spatial Expression for the spot-based ones.</p>
<p>We set up a class for spatial data analysis at single-cell/spot level providing ad-hoc attributes and specifically designed methods for attributes managing.</p>
<p>This latest version of the class is still under development even it’s already usable.</p>
</div>
<div id="extra-resources" class="section level1">
<h1 class="hasAnchor">
<a href="#extra-resources" class="anchor"></a>Extra Resources</h1>
<p>If you’re interested in a <strong>multimodal seqFISH dataset</strong> you can retrieve it from the <a href="https://bioconductor.org/packages/SpatialExperiment/"><strong>SingleCellMultiModal</strong> Bioconductor</a> package.</p>
<p>It is a multimodal dataset providing data for <em>seqFISH</em> and <em>scRNAseq</em> data, these data can be combined for multimodal single-cell data integration methods developments, testing and/or analysis.</p>
<p>More details about an investigation of this dataset can be found at this <a href="https://drighelli.github.io/BIRSBIO2020.seqFISH.SpatialAnalysis/articles/seqFISH.html">link</a>.</p>
<p>Or you can run:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"SingleCellMultiModal"</span><span class="op">)</span>

<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"SingleCellMultiModal"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/vignette.html">vignette</a></span><span class="op">(</span><span class="st">"seqFISH"</span>, package<span class="op">=</span><span class="st">"SingleCellMultiModal"</span><span class="op">)</span></code></pre></div>
</div>
<div id="references" class="section level1">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<ul>
<li><p>(seqFISH) Lubeck, E., Coskun, A., Zhiyentayev, T. et al. Single-cell in situ RNA profiling by sequential hybridization. Nat Methods 11, 360–361 (2014). <a href="https://doi.org/10.1038/nmeth.2892" class="uri">https://doi.org/10.1038/nmeth.2892</a></p></li>
<li><p>(seqFISH+) Eng, CH.L., Lawson, M., Zhu, Q. et al. Transcriptome-scale super-resolved imaging in tissues by RNA seqFISH+. Nature 568, 235–239 (2019). <a href="https://doi.org/10.1038/s41586-019-1049-y" class="uri">https://doi.org/10.1038/s41586-019-1049-y</a></p></li>
<li><p>(MERFISH) Chen, Kok Hao, et al. “Spatially resolved, highly multiplexed RNA profiling in single cells.” Science 348.6233 (2015).</p></li>
<li><p>(Slide-seqV2) Stickels, Robert R., et al. “Sensitive spatial genome wide expression profiling at cellular resolution.” bioRxiv (2020).</p></li>
<li><p>seqFISH reference website: <a href="https://www.seqfish.com/technology">https://www.seqfish.com/technology</a></p></li>
<li><p>10x Genomics Visium Spatial Gene Expression website <a href="https://www.10xgenomics.com/products/spatial-gene-expression">https://www.10xgenomics.com/products/spatial-gene-expression</a></p></li>
</ul>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Dario Righelli, Helena L Crowell, Lukas M Weber.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
