<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>3. The STexampleData package • SpatialExperimentBioc2021</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="3. The STexampleData package">
<meta property="og:description" content="SpatialExperimentBioc2021">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SpatialExperimentBioc2021</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Vig01_Introduction.html">1. Introduction to Spatial Transcriptomics</a>
    </li>
    <li>
      <a href="../articles/Vig02_SpatialExperiment.html">2. The SpatialExperiment class</a>
    </li>
    <li>
      <a href="../articles/Vig03_STexampleData.html">3. The STexampleData package</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/drighelli/SpatialExperiment_Bioc2021">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="Vig03_STexampleData_files/header-attrs-2.9/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>3. The STexampleData package</h1>
                        <h4 class="author">Lukas M. Weber</h4>
            <address class="author_afil">
      Johns Hopkins Bloomberg School of Public Health, Baltimore, USA<br><a class="author_email" href="mailto:#"></a><a href="mailto:lukas.weber@jhu.edu" class="email">lukas.weber@jhu.edu</a>
      </address>
                  
      
      <small class="dont-index">Source: <a href="https://github.com/drighelli/SpatialExperiment_Bioc2021/blob/master/vignettes/Vig03_STexampleData.Rmd"><code>vignettes/Vig03_STexampleData.Rmd</code></a></small>
      <div class="hidden name"><code>Vig03_STexampleData.Rmd</code></div>

    </div>

    
    
<div id="stexampledata-package" class="section level1">
<h1 class="hasAnchor">
<a href="#stexampledata-package" class="anchor"></a>STexampleData package</h1>
<p>The <a href="http://bioconductor.org/packages/STexampleData">STexampleData</a> package provides access to several spatially resolved transcriptomics datasets, which have been formatted into the <a href="https://bioconductor.org/packages/SpatialExperiment">SpatialExperiment</a> Bioconductor class.</p>
<p>These datasets have been collected from various publicly available sources, and cover several technological platforms.</p>
<p>We provide these datasets as <code>SpatialExperiment</code> objects to make them easier to access, so that we and others can use them for examples, demonstrations, tutorials, and other purposes.</p>
<p>Additional datasets from the 10x Genomics Visium platform are also available in the <a href="https://bioconductor.org/packages/TENxVisiumData">TENxVisiumData</a> package (developed by Helena Crowell).</p>
</div>
<div id="datasets" class="section level1">
<h1 class="hasAnchor">
<a href="#datasets" class="anchor"></a>Datasets</h1>
<p>The package currently contains the following datasets:</p>
<ul>
<li><p><code>Visium_humanDLPFC</code> (10x Genomics Visium): A single sample (sample 151673) of human brain dorsolateral prefrontal cortex (DLPFC) in the human brain, measured using the 10x Genomics Visium platform. This is a subset of the full dataset containing 12 samples from 3 neurotypical donors, published by <a href="https://www.nature.com/articles/s41593-020-00787-0">Maynard and Collado-Torres et al. (2021)</a>. The full dataset is available from the <a href="http://bioconductor.org/packages/spatialLIBD">spatialLIBD</a> Bioconductor package.</p></li>
<li><p><code>Visium_mouseCoronal</code> (10x Genomics Visium): A single coronal section from the mouse brain, spanning one hemisphere. This dataset was previously released by 10x Genomics on their <a href="https://support.10xgenomics.com/spatial-gene-expression/datasets/1.1.0/V1_Adult_Mouse_Brain">website</a>.</p></li>
<li><p><code>seqFISH_mouseEmbryo</code> (seqFISH): A subset of cells (embryo 1, z-slice 2) from a previously published dataset investigating mouse embryogenesis by <a href="https://www.biorxiv.org/content/10.1101/2020.11.20.391896v1">Lohoff and Ghazanfar et al. (2020)</a>, generated using the seqFISH platform. The full dataset is <a href="https://marionilab.cruk.cam.ac.uk/SpatialMouseAtlas/">available online</a>.</p></li>
</ul>
</div>
<div id="install-package" class="section level1">
<h1 class="hasAnchor">
<a href="#install-package" class="anchor"></a>Install package</h1>
<p>The package can be installed from Bioconductor as follows.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"STexampleData"</span>, version <span class="op">=</span> <span class="st">"devel"</span><span class="op">)</span></code></pre></div>
<p>Alternatively, the latest version can also be installed from GitHub. For this workshop, we use the latest development version from GitHub.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html">install_github</a></span><span class="op">(</span><span class="st">"lmweber/STexampleData"</span>, ref <span class="op">=</span> <span class="st">"no_accessors"</span>, build_vignettes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
</div>
<div id="load-dataset" class="section level1">
<h1 class="hasAnchor">
<a href="#load-dataset" class="anchor"></a>Load dataset</h1>
<p>First, we check the available datasets from the <code>STexampleData</code> package stored in Bioconductor’s ExperimentHub resource.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># load packages</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">ExperimentHub</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># query ExperimentHub resource</span>
<span class="va">eh</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ExperimentHub/man/ExperimentHub-class.html">ExperimentHub</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## snapshotDate(): 2021-08-04</code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">query</span><span class="op">(</span><span class="va">eh</span>, <span class="st">"STexampleData"</span><span class="op">)</span></code></pre></div>
<pre><code>## ExperimentHub with 6 records
## # snapshotDate(): 2021-08-04
## # $dataprovider: NA
## # $species: Mus musculus, Homo sapiens
## # $rdataclass: SpatialExperiment
## # additional mcols(): taxonomyid, genome, description,
## #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,
## #   rdatapath, sourceurl, sourcetype 
## # retrieve records with, e.g., 'object[["EH5443"]]' 
## 
##            title                   
##   EH5443 | Visium_humanDLPFC_3_13  
##   EH5444 | Visium_mouseCoronal_3_13
##   EH5445 | seqFISH_mouseEmbryo_3_13
##   EH6708 | Visium_humanDLPFC       
##   EH6709 | Visium_mouseCoronal     
##   EH6710 | seqFISH_mouseEmbryo</code></pre>
<p>Next, we run a short example showing how to load one of the datasets from the <code>STexampleData</code> package.</p>
<p>Alternative ways of loading the datasets from ExperimentHub are shown in the package vignette.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># load packages</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">SpatialExperiment</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/lmweber/STexampleData">STexampleData</a></span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># load dataset from latest development version</span>
<span class="va">spe</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/STexampleData/man/load_data.html">load_data</a></span><span class="op">(</span><span class="st">"Visium_humanDLPFC"</span><span class="op">)</span>

<span class="co"># alternatively: load from ExperimentHub using named accessor function</span>
<span class="co"># spe &lt;- Visium_humanDLPFC()</span></code></pre></div>
<p>Then, we have a look at the structure of the <code>SpatialExperiment</code> object.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">spe</span></code></pre></div>
<pre><code>## class: SpatialExperiment 
## dim: 33538 4992 
## metadata(0):
## assays(1): counts
## rownames(33538): ENSG00000243485 ENSG00000237613 ... ENSG00000277475
##   ENSG00000268674
## rowData names(3): gene_id gene_name feature_type
## colnames(4992): AAACAACGAATAGTTC-1 AAACAAGTATCTCCCA-1 ...
##   TTGTTTGTATTACACG-1 TTGTTTGTGTAAATTC-1
## colData names(3): cell_count ground_truth sample_id
## reducedDimNames(0):
## mainExpName: NULL
## altExpNames(0):
## spatialData names(6) : barcode_id in_tissue ... pxl_col_in_fullres
##   pxl_row_in_fullres
## spatialCoords names(2) : x y
## imgData names(4): sample_id image_id data scaleFactor</code></pre>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 33538  4992</code></pre>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">assayNames</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "counts"</code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">rowData</span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></code></pre></div>
<pre><code>## DataFrame with 33538 rows and 3 columns
##                         gene_id   gene_name    feature_type
##                     &lt;character&gt; &lt;character&gt;     &lt;character&gt;
## ENSG00000243485 ENSG00000243485 MIR1302-2HG Gene Expression
## ENSG00000237613 ENSG00000237613     FAM138A Gene Expression
## ENSG00000186092 ENSG00000186092       OR4F5 Gene Expression
## ENSG00000238009 ENSG00000238009  AL627309.1 Gene Expression
## ENSG00000239945 ENSG00000239945  AL627309.3 Gene Expression
## ...                         ...         ...             ...
## ENSG00000277856 ENSG00000277856  AC233755.2 Gene Expression
## ENSG00000275063 ENSG00000275063  AC233755.1 Gene Expression
## ENSG00000271254 ENSG00000271254  AC240274.1 Gene Expression
## ENSG00000277475 ENSG00000277475  AC213203.1 Gene Expression
## ENSG00000268674 ENSG00000268674     FAM231C Gene Expression</code></pre>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/SpatialExperiment/man/SpatialExperiment-colData.html">colData</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></code></pre></div>
<pre><code>## DataFrame with 4992 rows and 3 columns
##                    cell_count ground_truth     sample_id
##                     &lt;integer&gt;     &lt;factor&gt;   &lt;character&gt;
## AAACAACGAATAGTTC-1         NA       NA     sample_151673
## AAACAAGTATCTCCCA-1          6       Layer3 sample_151673
## AAACAATCTACTAGCA-1         16       Layer1 sample_151673
## AAACACCAATAACTGC-1          5       WM     sample_151673
## AAACAGAGCGACTCCT-1          2       Layer3 sample_151673
## ...                       ...          ...           ...
## TTGTTTCACATCCAGG-1          3       WM     sample_151673
## TTGTTTCATTAGTCTA-1          4       WM     sample_151673
## TTGTTTCCATACAACT-1          3       Layer6 sample_151673
## TTGTTTGTATTACACG-1         16       WM     sample_151673
## TTGTTTGTGTAAATTC-1          5       Layer2 sample_151673</code></pre>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/SpatialExperiment/man/SpatialExperiment-methods.html">spatialData</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></code></pre></div>
<pre><code>## DataFrame with 4992 rows and 6 columns
##                            barcode_id in_tissue array_row array_col
##                           &lt;character&gt; &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;
## AAACAACGAATAGTTC-1 AAACAACGAATAGTTC-1         0         0        16
## AAACAAGTATCTCCCA-1 AAACAAGTATCTCCCA-1         1        50       102
## AAACAATCTACTAGCA-1 AAACAATCTACTAGCA-1         1         3        43
## AAACACCAATAACTGC-1 AAACACCAATAACTGC-1         1        59        19
## AAACAGAGCGACTCCT-1 AAACAGAGCGACTCCT-1         1        14        94
## ...                               ...       ...       ...       ...
## TTGTTTCACATCCAGG-1 TTGTTTCACATCCAGG-1         1        58        42
## TTGTTTCATTAGTCTA-1 TTGTTTCATTAGTCTA-1         1        60        30
## TTGTTTCCATACAACT-1 TTGTTTCCATACAACT-1         1        45        27
## TTGTTTGTATTACACG-1 TTGTTTGTATTACACG-1         1        73        41
## TTGTTTGTGTAAATTC-1 TTGTTTGTGTAAATTC-1         1         7        51
##                    pxl_col_in_fullres pxl_row_in_fullres
##                             &lt;integer&gt;          &lt;integer&gt;
## AAACAACGAATAGTTC-1               2435               3913
## AAACAAGTATCTCCCA-1               8468               9791
## AAACAATCTACTAGCA-1               2807               5769
## AAACACCAATAACTGC-1               9505               4068
## AAACAGAGCGACTCCT-1               4151               9271
## ...                               ...                ...
## TTGTTTCACATCCAGG-1               9396               5653
## TTGTTTCATTAGTCTA-1               9630               4825
## TTGTTTCCATACAACT-1               7831               4631
## TTGTTTGTATTACACG-1              11193               5571
## TTGTTTGTGTAAATTC-1               3291               6317</code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SpatialExperiment/man/SpatialExperiment-methods.html">spatialCoords</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>##                       x    y
## AAACAACGAATAGTTC-1 3913 2435
## AAACAAGTATCTCCCA-1 9791 8468
## AAACAATCTACTAGCA-1 5769 2807
## AAACACCAATAACTGC-1 4068 9505
## AAACAGAGCGACTCCT-1 9271 4151
## AAACAGCTTTCAGAAG-1 3393 7583</code></pre>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/SpatialExperiment/man/SpatialExperiment-methods.html">imgData</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></code></pre></div>
<pre><code>## DataFrame with 2 rows and 4 columns
##       sample_id    image_id   data scaleFactor
##     &lt;character&gt; &lt;character&gt; &lt;list&gt;   &lt;numeric&gt;
## 1 sample_151673      lowres   ####   0.0450045
## 2 sample_151673       hires   ####   0.1500150</code></pre>
</div>
<div id="plot-dataset" class="section level1">
<h1 class="hasAnchor">
<a href="#plot-dataset" class="anchor"></a>Plot dataset</h1>
<p>Next, we generate some simple plots of the dataset. We can use plotting functions available in the <a href="https://github.com/lmweber/ggspavis">ggspavis</a> package, which is available from GitHub.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># install ggspavis package</span>
<span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html">install_github</a></span><span class="op">(</span><span class="st">"lmweber/ggspavis"</span>, build_vignettes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># load packages</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/lmweber/ggspavis">ggspavis</a></span><span class="op">)</span></code></pre></div>
<p>Create some plots of the dataset in x-y coordinates. This dataset contains a set of manually annotated ground truth labels for cortical layers in the human brain, which we can use to display annotation on the plot.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># plot spots only</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggspavis/man/plotSpots.html">plotSpots</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></code></pre></div>
<p><img src="Vig03_STexampleData_files/figure-html/unnamed-chunk-10-1.png" width="65%"></p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># plot spots with annotation</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggspavis/man/plotSpots.html">plotSpots</a></span><span class="op">(</span><span class="va">spe</span>, annotate <span class="op">=</span> <span class="st">"ground_truth"</span><span class="op">)</span></code></pre></div>
<p><img src="Vig03_STexampleData_files/figure-html/unnamed-chunk-11-1.png" width="65%"></p>
<p>We can also use a more flexible plotting function to display the spots and annotation along with the histology image as background, which is stored in the <code>SpatialExperiment</code> object. We also highlight spots located over tissue, which are identified with the “in_tissue” column in the <code>spatialData</code>.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># plots with histology background</span>
<span class="fu"><a href="https://rdrr.io/pkg/ggspavis/man/plotVisium.html">plotVisium</a></span><span class="op">(</span><span class="va">spe</span><span class="op">)</span></code></pre></div>
<p><img src="Vig03_STexampleData_files/figure-html/unnamed-chunk-12-1.png" width="65%"></p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/ggspavis/man/plotVisium.html">plotVisium</a></span><span class="op">(</span><span class="va">spe</span>, fill <span class="op">=</span> <span class="st">"ground_truth"</span>, highlight <span class="op">=</span> <span class="st">"in_tissue"</span><span class="op">)</span></code></pre></div>
<p><img src="Vig03_STexampleData_files/figure-html/unnamed-chunk-12-2.png" width="65%"></p>
</div>
<div id="exercise" class="section level1">
<h1 class="hasAnchor">
<a href="#exercise" class="anchor"></a>Exercise</h1>
<p>For some additional practice with these objects, attempt the following:</p>
<ul>
<li>Generate a plot in x-y coordinates with annotation showing total unique molecular identifier (UMI) counts</li>
<li>Generate a plot in x-y coordinates with annotation showing expression of a specific gene of interest, e.g. SNAP25, MOBP, or PCP4</li>
</ul>
<p>You can refer to the help files for the plotting functions with <code><a href="https://rdrr.io/pkg/ggspavis/man/plotSpots.html">?plotSpots</a></code> and <code><a href="https://rdrr.io/pkg/ggspavis/man/plotVisium.html">?plotVisium</a></code>.</p>
<p>For a further exercise, try adjusting the color palettes and other plotting options, as described in the help files.</p>
<!-- Possible solution 1 -->
<!-- colData(spe)$sum <- colSums(counts(spe)) -->
<!-- plotVisium(spe, fill = "sum", highlight = "in_tissue") -->
<!-- Possible solution 2 -->
<!-- id <- rowData(spe)$gene_id[which(rowData(spe)$gene_name == "PCP4")] -->
<!-- plotVisium(spe, fill = id, highlight = "in_tissue", assay = "counts") -->
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Dario Righelli, Helena L Crowell, Lukas M Weber.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
